src/function.ts

```diff
@@ -383,15 +387,30 @@ abstract class FunctionBase<in Payload, Out>
   public asl(call: CallExpr, context: ASL) {
     const payloadArg = call.getArgument("payload")?.expr;
     this.resource.grantInvoke(context.role);
+    // TODO generalize this?
+    const props = ((): Partial<Task> => {
+      if (!payloadArg) {
+        return {
+          Parameters: undefined,
+        };
+      } else if (isVariableReference(payloadArg)) {
+        return {
+          InputPath: ASL.toJsonPath(payloadArg),
+        };
+      } else if (isStringLiteralExpr(payloadArg)) {
+        return {
+          Parameters: payloadArg.value,
+        };
+      } else {
+        return {
+          Parameters: ASL.toJson(payloadArg),
+        };
+      }
+    })();
```

Comment:
Use ternary instead of an instantly called function closure?

Reply:
I can, yeah. This will likely get much more complex and at one point I was trying to generate multiple states. Right now I guess it would be a mostly readable ternary.